### PR TITLE
Integrate gutenberg-mobile release 1.72.0

### PR DESCRIPTION
**Pull request by jhnstn**
Description
This PR incorporates the 1.72.0 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR:

https://github.com/wordpress-mobile/gutenberg-mobile/pull/4668

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to \`RELEASE-NOTES.txt\` if necessary."

### CODE REVIEW

File: Podfile
Comment (hypest) [inline Podfile]: Should we update to point to https://github.com/wordpress-mobile/gutenberg-mobile/commit/8b29453cdbd5547bb666ab83bf60530f3024ce60 that includes the bundle update among other things? 🤔

Comment (jhnstn) [reply Podfile]: `2095f19d1a7` should be fine since it is ahead of `8b29453cdb`, That is gutenberg-mobile @`2095f19d1a7` includes all the changes in  `8b29453cdb`

Comment (jhnstn) [reply Podfile]: Just for completeness, `2095f19d1a7` is not the current head of the Gutenberg Mobile release PR. But `2095f19d1a7` just behind some release note changes so there are no behavioral changes.l

Comment (hypest) [reply Podfile]: Oh, you mean that the bundle change in https://github.com/wordpress-mobile/gutenberg-mobile/pull/4668/commits/6b030ad4fac4a9053dc854307852323e38b7a2e0 is not really a functional change, just changelog changes on the Gutenberg side?

Comment (jhnstn) [reply Podfile]: Nope. `8b29453` , `6b030ad`, and `2095f19d1` all are from the release script, also in that specific order. `2095f19d1`, being the last commit by the release script, includes all the changes from `6b030ad` and  `8b29453`. 

My last comment was just noting that `2095f19d1` is no longer the head of the GB mobile release branch. But all the later commits are just change log updates. Some are coming from Gutenberg so there are some bundle diffs but again, anything after `2095f19d1` is non behavioral

Comment (hypest) [reply Podfile]: Gotcha, thanks for elaborating Jason! 🙇‍♂️

**Review by hypest — APPROVED** [review]
I will LGTM the PR @jhnstn as the changes make sense to me, and to enable you to move forward with the merge domino. The only thing to ask is to update the PRs so all pointers are the latest and wait for all-green in all PRs before the merge domino start. Thanks!

**Pull request merged**
